Google Books.js

```diff
@@ -116,17 +116,22 @@ function parseXML(text) {
 	}
 	
 	var ISBN;
-	const ISBN10Re = /(ISBN:)(\w{10})$/;
-	const ISBN13Re = /(ISBN:)(\w{13})$/;
+	const ISBN10Re = /(?:ISBN:)(\w{10})$/;
+	const ISBN13Re = /(?:ISBN:)(\w{13})$/;
+	const booksIDRe = /^(\w{12})$/;
 	var identifiers = ZU.xpath(xml, "dc:identifier", ns);
 	for (var i in identifiers) {
 		var ISBN10Match = ISBN10Re.exec(identifiers[i].textContent);
 		var ISBN13Match = ISBN13Re.exec(identifiers[i].textContent);
+		var booksIDMatch = booksIDRe.exec(identifiers[i].textContent);
 		if (ISBN10Match != null) {
-			ISBN = ISBN10Match[2];
+			ISBN = ISBN10Match[1];
 		}
 		if (ISBN13Match != null) {
-			ISBN = ISBN13Match[2];
+			ISBN = ISBN13Match[1];
+		}
+		if (booksIDMatch != null) {
+			newItem.url = 'https://books.google.com/books?id=' + booksIDMatch[1];
```

Comment:
I don't mean just for Wikipedia in this case, and perhaps it's more for "book section" than "book"-type citations, but I'll often have a passage of one to ten pages that I'd cite on its own from a chapter and if I recall correctly, the Zotero ingest doesn't take the digital webpage into account. I would think that the point of having full-page text readable online is to be able to share links to passages like that and thus to have them in the citations. I agree that I rarely see books.google.com URLs in bibliographies for books, and I can't say I recall seeing it often for chapters or passages either. For me this is more of keeping my research in the same place. Apart from that, no qualms in particular about using the `extra` field as you mentioned for citation generation.